### PR TITLE
Fix for checkFFDC failures on Java 15

### LINKED ISSUE #13946: [Java15] Test Failure (20200914-0243): com.ibm.ws.logging.json.fat.MessagesLogEnvTest.checkFfdc
RTC 277839

```
junit.framework.AssertionFailedError: 2020-09-14-02:04:46:598 Test failed with one or more errors
at com.ibm.ws.logging.json.fat.JSONEventsTest.checkFfdc(JSONEventsTest.java:118)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:197)
at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:318)
at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:171)
```

### PR DESCRIPTION
**Pull request by Yushan-Lin**
Fixes #13946 
Java 15 has additional field "message" in the ffdc log.

### CODE REVIEW

**Comment by Yushan-Lin** [comment]
#build

**Comment by Yushan-Lin** [comment]
Java 15: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_LnxVMPhKEeqwgKOHPUEi3Q

**Comment by fmhwong** [comment]
@Yushan-Lin Do you know why we have a message field in just Java 15?  What does it contain?

**Comment by Yushan-Lin** [comment]
@fmhwong on Java 15 the NPE thrown outputs an ffdc log with the message field:
`"message":"Cannot invoke \"String.toString()\" because \"myString\" is null"`

**Comment by Yushan-Lin** [comment]
@fmhwong For Java 15, the exception thrown has a message `"message":"Cannot invoke \"String.toString()\" because \"myString\" is null"` while lower Java versions does not have the message. In FFDC JSON logs, the message field will appear as long as the message is not null.

**Review by fmhwong — CHANGES_REQUESTED** [review]
See comments

File: dev/com.ibm.ws.logging.json_fat/fat/src/com/ibm/ws/logging/json/fat/JSONEventsTest.java [hunk 1]
```diff
@@ -101,7 +101,8 @@ public void checkFfdc() throws Exception {
                                                                                       "ibm_sequence", "ibm_className", "ibm_exceptionName", "ibm_probeID",
                                                                                       "ibm_threadId", "ibm_stackTrace", "ibm_objectDetails"));
 
-        ArrayList<String> ffdcKeysOptionalList = new ArrayList<String>();
+        //JDK 15 has message as an additional field in ffdc events
+        ArrayList<String> ffdcKeysOptionalList = new ArrayList<String>(Arrays.asList("message"));
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging.json_fat/fat/src/com/ibm/ws/logging/json/fat/JSONEventsTest.java:105]: I am suggesting to keep `checkFfdc()` for Java 14 and lower.  Create a new method for Java 15 and make the message field mandatory.

**Comment by Yushan-Lin** [comment]
#build

**Comment by fmhwong** [comment]
@Yushan-Lin Can you request a Java15 build?

**Comment by Yushan-Lin** [comment]
Java 15: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_iOgHgP2aEeqZosoAJ-GPIQ

**Review by fmhwong — APPROVED** [review]
LGTM

**Comment by Yushan-Lin** [comment]
#run-libby-bot

**Pull request merged**